JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeScreen.kt

```diff
@@ -121,103 +139,160 @@ fun HomeScreen(
+            if (uiState.errorMessages.isNotEmpty()) {
+                // Remember the errorMessage to display on the screen
+                val errorMessage = remember(uiState) { uiState.errorMessages[0] }
 
-        // Get the text to show on the message from resources
-        val errorMessageText: String = stringResource(errorMessage.messageId)
-        val retryMessageText = stringResource(id = R.string.retry)
+                // Get the text to show on the message from resources
+                val errorMessageText: String = stringResource(errorMessage.messageId)
+                val retryMessageText = stringResource(id = R.string.retry)
 
-        // If onRefreshPosts or onErrorDismiss change while the LaunchedEffect is running,
-        // don't restart the effect and use the latest lambda values.
-        val onRefreshPostsState by rememberUpdatedState(onRefreshPosts)
-        val onErrorDismissState by rememberUpdatedState(onErrorDismiss)
+                // If onRefreshPosts or onErrorDismiss change while the LaunchedEffect is running,
+                // don't restart the effect and use the latest lambda values.
+                val onRefreshPostsState by rememberUpdatedState(onRefreshPosts)
+                val onErrorDismissState by rememberUpdatedState(onErrorDismiss)
 
-        // Effect running in a coroutine that displays the Snackbar on the screen
-        // If there's a change to errorMessageText, retryMessageText or scaffoldState,
-        // the previous effect will be cancelled and a new one will start with the new values
-        LaunchedEffect(errorMessageText, retryMessageText, scaffoldState) {
-            val snackbarResult = scaffoldState.snackbarHostState.showSnackbar(
-                message = errorMessageText,
-                actionLabel = retryMessageText
-            )
-            if (snackbarResult == SnackbarResult.ActionPerformed) {
-                onRefreshPostsState()
+                // Effect running in a coroutine that displays the Snackbar on the screen
+                // If there's a change to errorMessageText, retryMessageText or scaffoldState,
+                // the previous effect will be cancelled and a new one will start with the new values
+                LaunchedEffect(errorMessageText, retryMessageText, scaffoldState) {
+                    val snackbarResult = scaffoldState.snackbarHostState.showSnackbar(
+                        message = errorMessageText,
+                        actionLabel = retryMessageText
+                    )
+                    if (snackbarResult == SnackbarResult.ActionPerformed) {
+                        onRefreshPostsState()
+                    }
+                    // Once the message is displayed and dismissed, notify the ViewModel
+                    onErrorDismissState(errorMessage.id)
+                }
+            }
+        } else {
+            val selectedPost = uiState.posts.find { it.id == uiState.selectedPostId }
```

Comment:
Better if this logic goes to the VM and we expose the actual post and not just the ID

Reply:
This changed a bit with the latest version, let me know what you think!